### PR TITLE
terraform 0.4.1

### PR DESCRIPTION
**Pull request by phinze**
Wow this was _non_-trivial!

I've got a script to automate bits of the dependency list - I'll try to see if maybe I can automate the generation of the formula from the terraform side so we can do this when we release.

In the meantime let me know if I missed anything here. :+1: :grinning:

### CODE REVIEW

File: Library/Formula/terraform.rb [hunk 1]
```diff
@@ -23,72 +32,104 @@ class Terraform < Formula
...
-  go_resource "github.com/mitchellh/iochan" do
-    url "https://github.com/mitchellh/iochan.git", :revision => "b584a329b193e206025682ae6c10cdbe03b0cd77"
+  go_resource "code.google.com/p/go-uuid" do
+    url "https://code.google.com/p/go-uuid", :revision => "35bc42037350f0078e3c974c6ea690f1926603ab", :using => :hg
   end
 
-  go_resource "github.com/hashicorp/hcl" do
-    url "https://github.com/hashicorp/hcl.git", :revision => "513e04c400ee2e81e97f5e011c08fb42c6f69b84"
+  go_resource "google.golang.org/api" do
+    url "https://code.googlesource.com/api.git", :revision => "2f6114897375589857c508d7392e55d5e7580db8"
```
Comment (mikemcquaid) [inline Library/Formula/terraform.rb]: Looks like this no longer exists: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24015/version=yosemite/testReport/junit/brew-test-bot/yosemite/fetch_terraform/

Comment (phinze) [inline Library/Formula/terraform.rb]: Heh, that was a fast change! Investigating...

Comment (phinze) [inline Library/Formula/terraform.rb]: Ah, turns out I was leaning on caches locally from several subsequent runs.

**Comment by phinze** [comment]
Okay I think this passes the build now. I may have confused Jenkins it with a couple of poorly timed `push -f`s to my branch.

This is my first real interaction with the `test-bot` and let me say IT IS AMAZING. :heart_eyes_cat: Great work @mikemcquaid, @mistydemeo, and co!

**Comment by mikemcquaid** [comment]
@phinze It would probably be relatively easy to get the bot testing `cask` too if that's something you were ever interested in.

**Comment by phinze** [comment]
> @phinze It would probably be relatively easy to get the bot testing cask too if that's something you were ever interested in.

@mikemcquaid I am definitely definitely interested in exploring this. I'll reach out to you sometime soon to chat about it. :+1:

**Comment by mikemcquaid** [comment]
@phinze Relatedly I'd also love to discuss moving Homebrew and Cask closer together at some point so that you can be more "Official".

File: Library/Homebrew/boxen-bottle-hooks.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Bottle hooks for Homebrew.
```
Comment (mikemcquaid) [inline Library/Homebrew/boxen-bottle-hooks.rb]: Whoops!

Comment (mikemcquaid) [inline Library/Homebrew/boxen-bottle-hooks.rb]: I'll remove them locally.

Comment (phinze) [inline Library/Homebrew/boxen-bottle-hooks.rb]: :see_no_evil: Need more :coffee:

**Comment by matthewford** [comment]
@phinze think we could get terraform 0.4.2 in? :wink:

**Comment by mikemcquaid** [comment]
@matthewford Could you try and open a pull request? This should help and I'm happy to walk you through anything else: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged

Thanks!

**Pull request closed without merging**